### ISSUE TITLE
[Python] Add a FixedSizeTensorScalar class

### ISSUE THREAD

**Issue by wjones127** [apache/arrow#37484]
### Describe the enhancement requested

When you get a scalar off of a `FixedSizeTensorArray`, you get back a generic `ExtensionTensor`. There's not an easy way to get the tensor contained back as a numpy ndarray. It would be nice to have:

```python
def FixedSizeTensorScalar(pa.ExtensionScalar):
    def to_numpy_ndarray(): ...
```

Then one could do:

```python
tensor_array[i].to_numpy_ndarray()
```

As a workaround, I currently do:

```python
tensor_array.to_numpy_ndarray()[i]
```

Which isn't too bad.

### Component(s)

Python

**Comment by jorisvandenbossche** [comment]
Sounds good!

**Comment by rok** [comment]
I can grab this one.

Also, another workaround would be:
```
tensor_array.slice(i, i+1).to_numpy_ndarray()[0]
```

**Comment by wjones127** [comment]
> Also, another workaround would be:
> tensor_array.slice(i, i+1).to_numpy_ndarray()[0]

Yes, but that doesn't work in 12.0.0. There is a bug that was fixed in 13.0.0 where `to_numpy_ndarray()` didn't work properly on sliced tensors.

**Comment by rok** [comment]
Oh yeah, that's right.
